test/bdd/exec_test.go

```diff
@@ -0,0 +1,334 @@
+package bdd
+
+/*
+Copyright 2019 LitmusChaos Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appv1 "k8s.io/api/apps/v1"
+	//batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	scheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/litmuschaos/chaos-operator/pkg/apis/litmuschaos/v1alpha1"
+	chaosClient "github.com/litmuschaos/chaos-operator/pkg/client/clientset/versioned/typed/litmuschaos/v1alpha1"
+	restclient "k8s.io/client-go/rest"
+)
+
+var (
+	kubeconfig string
+	config     *restclient.Config
+	client     *kubernetes.Clientset
+	clientSet  *chaosClient.LitmuschaosV1alpha1Client
+)
+
+func init() {
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "path to kubeconfig to invoke kubernetes API calls")
+}
+func TestChaos(t *testing.T) {
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "BDD test")
+}
+
+var _ = BeforeSuite(func() {
+
+	var err error
+	kubeconfig = os.Getenv("HOME") + "/.kube/config"
+	config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+
+	if err != nil {
+		Expect(err).To(BeNil(), "failed to get config")
+	}
+
+	client, err = kubernetes.NewForConfig(config)
+
+	if err != nil {
+		Expect(err).To(BeNil(), "failed to get client")
+	}
+
+	clientSet, err = chaosClient.NewForConfig(config)
+
+	if err != nil {
+		Expect(err).To(BeNil(), "failed to get clientSet")
+	}
+
+	err = v1alpha1.AddToScheme(scheme.Scheme)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	//Creating crds
+	By("creating chaosengine crd")
+	err = exec.Command("kubectl", "create", "-f", "../../deploy/chaos_crds.yaml").Run()
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	//Creating rbacs
+	err = exec.Command("kubectl", "create", "-f", "../../deploy/rbac.yaml").Run()
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	//Creating Chaos-Operator
+	By("creating operator")
+	err = exec.Command("kubectl", "create", "-f", "../../deploy/operator.yaml").Run()
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	fmt.Println("chaos-operator created successfully")
+
+	//Wait for the creation of chaos-operator
+	time.Sleep(10 * time.Second)
```

Comment:
Is this sleep required? anyway we have the check in next line

Reply:
yes, because of the check, we have to wait.

Reply:
Instead of sleep can we use something like, Retry in the loop for 5-6 times then fail